diff.py

```diff
@@ -1772,7 +1805,7 @@ class ArchSettings:
     re_reg=re.compile(r"\$?\b([rf][0-9]+)\b"),
     re_sprel=re.compile(r"(?<=,)(-?[0-9]+|-?0x[0-9a-f]+)\(r1\)"),
     re_large_imm=re.compile(r"-?[1-9][0-9]{2,}|-?0x[0-9a-f]{3,}"),
-    re_imm=re.compile(r"(\b|-)([0-9]+|0x[0-9a-fA-F]+)\b(?!\(r1)|[^@]*@(ha|h|lo)"),
+    re_imm=re.compile(r"(\b|-)([0-9]+|0x[0-9a-fA-F]+)\b(?!\(r1)|[^@]*@(ha|h|lo|sda21)"),
```

Comment:
The `[^@]` here is very dubious btw, given that symbol names can contain `@`. It's probably what causes `normalize_imms` to break. Ideally we wouldn't include symbol names in the asm and then pattern match on that; given compiler-generated names with `@` and C++ mangled names it seems a recipe for failure. Replacing the `[^@]` by some other symbol separator could work though. For later, I guess...

Reply:
>The [^@] here is very dubious btw

I didn't do anything with that or this regex or uses of this regex. I just added the sda21 because you asked me to.

>Replacing the [^@] by some other symbol separator could work though. For later, I guess...

I will start working on this right away, it will be first priority for me (other than fixes to other PRs based on your review)
But it seems like the [^@] is not the main issue, the cases where I see it really breaking do not even include @ symbols.

I'll start digging into and report what I find.